### DISCUSSION TITLE
USB devices - GO Launchpad

### DISCUSSION THREAD

**Discussion by mattrixed** [GrandOrgue/grandorgue#1018] [Developers]
Hello knowledgeable programmers....! :)

Runnings latest 3.6.1-1
Main Issue 1:

Ever since a while ago when all the usb devices names got changed and we had to re-learn connected devices and they then had winmm prefixes etc I gave up on my launchpad with my setup as it continually crashed GO and other issues.  I've since figured out it was potentially adding to the issue a usb cable and now have it working however because connected usb devices no longer have a hub type id? common name when you right click a stop and assigning it to a button on launchpad it comes up as "any device" instead of listing the launchpad - this then causes the issue that its listening to it on channel 1 but I have a manual on channel 1 from the usb connected 18i20 via midi 5din Yamaha hs8 and when you press notes it changes all the stops assigned!! - one way around this is add to the launchpad in GO the offset midi 4 to make it channel 5 in midi.  But now it doesn't have its own device hub id? - you used to be able to have everything on channel 1 if you wished via different connected usb devices and they would be all independent because of seperate usb/midi connection.  But now channel 1 for eg affects everything like its all using old 5din connection with in/out multiple devices daisy connected...  (someone else talking of this perhaps on Facebook GO group recently)

I note that the focusrite comes up as a proper device listed on the manuals etc 
![IMG_3296](https://user-images.githubusercontent.com/84826875/152676944-c97cc704-0eae-4429-911e-68f67a4914a0.jpg)

Whilst the launchpad works and even sends data for colours buttons on off etc - change any device to any of the many listed launchpad devices disables the launchpad and it ONLY works when leaving it as detected "any device". - 

![IMG_3295](https://user-images.githubusercontent.com/84826875/152676929-987248b4-3741-4971-871e-950dadcdf4df.jpg)

Any ideas before I go back and open every single stop :-( and change the midi channel to 5?>!  Doing this fixes the problem as shown in the pic of working properly but that's another strange thing the launchpad is set to midi channel 6 by default but GO is sending note triggers on 1....?> this maybe something more to do with users and me not understanding how that works in the launchpad...

Issue 2:

When using multichannel for me for a very long time the meter bars up top don't correctly reflect the channels in order - I have 5 stereo pairs and it displays 5 but the first one is NOT channel 1/2! but rather 9/10 then 1/2 etc

Issue/Upgrade 3:

Extend the tails options instead of max and 50 - 3000ms could it be 50-5000 for eg - for long smaplesets might not want to restrict to 3 seconds might just want tail off a tad like 4 or 5 secs from a 6-7 max natural full.

Thankyou all - I've been riding the ups and downs with the app for a very long time, probably since late v2 v3 beginning on the old site - its a great app and have it configured on lots of different machines and use it every week with Freisach on pc and Mac.  I've given up on linux as can't get it installed like the good old days of typing in terminal code from a repository and don't have enough smarts to get a touch screen via serial to work on linux!   Would be fun to install again tho really need detailed instructions from a vanilla linux as it always comes up with missing dependancies etc... linux install GO for dummies!!!! with terminal commands and full links needed!!!  :-)

Regards,
Matt

**Comment by oleg68** [comment]
> Main Issue 1:

It is the same issue as #1000. It has already been fixed, but there is no any GO release yet. You may try to use [the intermediate build 3.6.1-1.3](https://github.com/GrandOrgue/grandorgue/actions/runs/1799671703) with this fix.

**Comment by oleg68** [comment]
> Issue 2:
> 
> When using multichannel for me for a very long time the meter bars up top don't correctly reflect the channels in order - I have 5 stereo pairs and it displays 5 but the first one is NOT channel 1/2! but rather 9/10 then 1/2 etc

Please create an issue. It'is possible not to display meters for not used channels (without audio groups assigned to).

**Comment by oleg68** [comment]
> Issue/Upgrade 3:
> 
> Extend the tails options instead of max and 50 - 3000ms could it be 50-5000 for eg - for long smaplesets might not want to restrict to 3 seconds might just want tail off a tad like 4 or 5 secs from a 6-7 max natural full.

Please create an issue.

**Comment by oleg68** [comment]
> I've given up on linux as can't get it installed like the good old days of typing in terminal code from a repository and don't have enough smarts to get a touch screen via serial to work on linux! Would be fun to install again tho really need detailed instructions from a vanilla linux as it always comes up with missing dependancies etc... linux install GO for dummies!!!! with terminal commands and full links needed!!! :-)

There are several ways for installing GO on Linux. Please take a look at [INSTALL.md](https://github.com/GrandOrgue/grandorgue/blob/master/INSTALL.md) for the instructions.

If you have only one touchscreen then it is usually works out of box. If you have several touchscreens please read [this link](https://github.com/GrandOrgue/grandorgue/discussions/736)
